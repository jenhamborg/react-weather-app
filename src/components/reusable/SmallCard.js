import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const CardContainer = styled.div`
  display: flex;
  align-items: center;
  flex-direction: row;
  background-color: rgba(255, 255, 255, 0.95);
  border-radius: 4px;
  color: #242526;
  margin-bottom: 1rem;
  text-align: center;
  height: 125px;
  width: 300px;
  h3 {
    font-size: 2rem;
  }
  .sm-card-col-one {
    text-align: center;
    margin-right: 0.5rem;
    width: 50%;
  }
  .sm-card-col-two {
    text-align: center;
    margin-left: 0.5rem;
    width: 50%;
  }
  .sm-card-vl {
    margin-bottom: auto;
    margin-top: auto;
    border-left: 3px solid #d3d3d3;
    height: 100px;
  }
  .sm-card-row-one {
    font-size: 2rem;
    font-weight: 700;
  }
  .sm-card-row-two {
    font-size: 1.25rem;
  }
`;

export default function SmallCard({
  columnOneRowOne,
  columnOneRowTwo,
  columnTwoRowOne,
  columnTwoRowTwo,
}) {
  return (
    <CardContainer>
      <div className="sm-card-col-one">
        <div className="sm-card-row-one">{columnOneRowOne}</div>
        <div className="sm-card-row-two">{columnOneRowTwo}</div>
      </div>
      <div className="sm-card-vl"></div>
      <div className="sm-card-col-two">
        <div className="sm-card-row-one">{columnTwoRowOne}</div>
        <div className="sm-card-row-two">{columnTwoRowTwo}</div>
      </div>
    </CardContainer>
  );
}

SmallCard.propTypes = {
  columnOneRowOne: PropTypes.string,
  columnOneRowTwo: PropTypes.string,
  columnTwoRowOne: PropTypes.string,
  columnTwoRowTwo: PropTypes.string,
};
