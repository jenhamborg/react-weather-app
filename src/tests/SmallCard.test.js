import React from "react";
import { render, cleanup } from "@testing-library/react";
import SmallCard from "../components/reusable/smallCard";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<SmallCard />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("Small Card", () => {
  afterEach(() => {
    cleanup;
  });

  it("Correct row data is displayed", async () => {
    const { getByText } = render(
      <SmallCard
        columnOneRowOne={"Apple"}
        columnOneRowTwo={"Orange"}
        columnTwoRowOne={"Strawberry"}
        columnTwoRowTwo={"Blueberry"}
      />
    );

    const contentOne = getByText("Apple");
    const contentTwo = getByText("Orange");
    const contentThree = getByText("Strawberry");
    const contentFour = getByText("Blueberry");

    expect(contentOne).toHaveTextContent("Apple");
    expect(contentTwo).toHaveTextContent("Orange");
    expect(contentThree).toHaveTextContent("Strawberry");
    expect(contentFour).toHaveTextContent("Blueberry");
  });
});
