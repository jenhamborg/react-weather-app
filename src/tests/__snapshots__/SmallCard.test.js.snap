// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="smallCard__CardContainer-sc-1smmpz4-0 eNOGhc"
>
  <div
    className="sm-card-col-one"
  >
    <div
      className="sm-card-row-one"
    />
    <div
      className="sm-card-row-two"
    />
  </div>
  <div
    className="sm-card-vl"
  />
  <div
    className="sm-card-col-two"
  >
    <div
      className="sm-card-row-one"
    />
    <div
      className="sm-card-row-two"
    />
  </div>
</div>
`;
